### PR TITLE
fix: Preventing recreation of pages when navigating back 

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): https://github.com/unoplatform/uno.chefs/issues/105

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

When navigating back with a return value the previous page was being recreated, even if NavigationCacheMode set to Required (on Android)

[Also, on Xamarin Android an exception is raised when using Console logging]

## What is the new behavior?

Removed code that was modifying the backstack - this is *only* used when retrieving the response value in the NavigatedTo event in the previous page - it's not used by Uno.Ext.Navigation, so removing this will not affect existing functionality.

[Also, removed console logging for Xamarin targets, and fixed build for the Legacy targets for testharness]

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../README.md#supported)
- [ N/A ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ N/A ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [X] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ N/A  ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: testing/TestHarness/TestHarness.Shared/Ext/Core/Storage/StorageOneViewModel.cs [hunk 1]
```diff
@@ -105,8 +106,8 @@ public async Task PerfTest()
 			var keyValues = new List<(string Key, string Value)>();
 			for (int i = 0; i < keyCount; i++)
 			{
-				var key = GenerateString(Random.Shared.Next(0, 100));
-				var value = GenerateString(Random.Shared.Next(0, 1000));
+				var key = GenerateString(_random.Next(0, 100));
+				var value = GenerateString(_random.Next(0, 1000));
```
Comment (jeromelaban) [inline testing/TestHarness/TestHarness.Shared/Ext/Core/Storage/StorageOneViewModel.cs:110]: I'm curious, why did you change this?

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.Extensions.Navigation.UI/Navigators/FrameNavigator.cs [hunk 2]
```diff
@@ -265,17 +265,6 @@ private async void FrameGoBack(object? parameter, RouteInfo? previousMapping)
...
-			{
-				if (Logger.IsEnabled(LogLevel.Debug)) Logger.LogDebugMessage($"Replacing last backstack item to inject parameter '{parameter.GetType().Name}'");
-				// If a parameter is being sent back, we need to replace
-				// the last frame on the backstack with one that has the correct
-				// parameter value. This value can be extracted via the OnNavigatedTo method
-				var entry = Control.BackStack.Last();
-				var newEntry = new PageStackEntry(entry.SourcePageType, parameter, entry.NavigationTransitionInfo);
-				Control.BackStack.Remove(entry);
-				Control.BackStack.Add(newEntry);
-			}
```
Comment (kazo0) [inline src/Uno.Extensions.Navigation.UI/Navigators/FrameNavigator.cs:278]: Should we keep this here for Windows at least? Or you think its better to have the behavior the same for all platforms?

**Review by kazo0 — APPROVED** [review]

File: src/Uno.Extensions.Navigation.UI/Navigators/FrameNavigator.cs (hunk 2 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Navigators/FrameNavigator.cs:278]: should have same behaviour on all platforms I think

File: testing/TestHarness/TestHarness.Shared/Ext/Core/Storage/StorageOneViewModel.cs (hunk 1 above)
Comment (nickrandolph) [reply testing/TestHarness/TestHarness.Shared/Ext/Core/Storage/StorageOneViewModel.cs:110]: Random.Shared isn't supported on Xamarin targets
